File: helparty/src/main/java/com/hamryt/helparty/service/gym/GymService.java
package com.hamryt.helparty.service.gym;

import com.hamryt.helparty.dto.gym.GymDTO;
import com.hamryt.helparty.dto.gym.request.SignUpGymRequest;
import com.hamryt.helparty.dto.gym.request.UpdateGymRequest;
import com.hamryt.helparty.dto.gym.response.SignUpGymResponse;
import com.hamryt.helparty.dto.gym.response.UpdateGymResponse;

public interface GymService {
    
    SignUpGymResponse insertGym(SignUpGymRequest resource);
    
    GymDTO findGymByEmailAndPassword(String email, String encryptPassword);
    
    UpdateGymResponse updateGym(Long loginId, UpdateGymRequest updateGymRequest);
    
    String findGymEmailById(Long id);
    
    void deleteGym(Long loginId);
}

File: src/main/java/com/hamryt/helparty/service/gym/GymService.kt
package com.hamryt.helparty.service.gym

import com.hamryt.helparty.dto.gym.GymDTO
import com.hamryt.helparty.dto.gym.request.SignUpGymRequest
import com.hamryt.helparty.dto.gym.request.UpdateGymRequest
import com.hamryt.helparty.dto.gym.response.SignUpGymResponse
import com.hamryt.helparty.dto.gym.response.UpdateGymResponse

interface GymService {

    fun insertGym(resource: SignUpGymRequest): SignUpGymResponse

    fun findGymByEmailAndPassword(email: String, password: String): GymDTO

    fun updateGym(loginId: Long, updateGymRequest: UpdateGymRequest): UpdateGymResponse

    fun deleteGym(loginId: Long)
}